File: ocaml/runsels.ml
(** The main executable. *)

open Support;;

let () =
  try
    let config = Config.get_default_config () in
    match List.tl (Array.to_list Sys.argv) with
    | [] -> failwith "usage: runsels selections.xml arg..."
    | (app_or_sels :: args) ->
        let sels_path = match Apps.lookup_app app_or_sels config with
        | None -> app_or_sels
        | Some app_path -> app_path +/ "selections.xml" in
        let sels = Qdom.parse_file sels_path in
        Run.execute_selections sels args config
  with
  | ex ->
      output_string stderr (Printexc.to_string ex);
      output_string stderr "\n";
      if not (Printexc.backtrace_status ()) then
        output_string stderr "(hint: run with OCAMLRUNPARAM=b to get a stack-trace)\n"
      else
        Printexc.print_backtrace stderr;
      exit 1
;;

File: python/runsels.py
#!/usr/bin/python2
import sys
from os.path import join

import qdom, run, apps
from basedirs import get_default_config

config = get_default_config()
args = sys.argv[1:]
if not args:
	raise Exception("Syntax: runsels.py [APP | SELECTIONS] [ARGS]")

app_or_sels = args[0]
app_args = args[1:]

app_mgr = apps.AppMgr(config)

app_path = app_mgr.lookup_app(app_or_sels, missing_ok = True)
sels_path = join(app_path, "selections.xml") if app_path is not None else app_or_sels

with open(sels_path, 'rb') as stream:
	sels = qdom.parse(stream)

runner = run.Runner(config)
runner.execute_selections(sels, app_args)
